### PR TITLE
PP-9203: Add ability to change a user's email address

### PR DESCRIPTION
**Pull request by oswaldquek**

### CODE REVIEW

File: src/main/java/uk/gov/pay/adminusers/service/UserServices.java [hunk 1]
```diff
@@ -267,6 +277,12 @@ private void changeUserTelephoneNumber(UserEntity userEntity, String telephoneNu
         userEntity.setUpdatedAt(ZonedDateTime.now(ZoneId.of("UTC")));
         userDao.merge(userEntity);
     }
+    
+    private void changeUserEmail(UserEntity userEntity, String email) {
+        userEntity.setEmail(email);
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/adminusers/service/UserServices.java:282]: I think we'd also want to update the username, which is, for some reason, a different field.

Comment (oswaldquek) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java:282]: Uh why? We're not updating the username but the email address

File: src/main/java/uk/gov/pay/adminusers/validations/UserPatchValidations.java [hunk 2]
```diff
@@ -21,11 +22,14 @@
...
+    private static final List<String> PATCH_ALLOWED_PATHS = 
+            List.of(PATH_SESSION_VERSION, PATH_DISABLED, PATH_TELEPHONE_NUMBER, PATH_FEATURES, PATH_EMAIL);
+    
     private static final Multimap<String, String> USER_PATCH_PATH_OPS = new ImmutableListMultimap.Builder<String, String>()
             .put(PATH_SESSION_VERSION, "append")
             .put(PATH_DISABLED, "replace")
             .put(PATH_TELEPHONE_NUMBER, "replace")
+            .put(PATH_EMAIL, "replace")
             .put(PATH_FEATURES, "replace")
             .build();
```
[the remark this answers is not in the document]
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/adminusers/validations/UserPatchValidations.java]: Good shout, [fixed](https://github.com/alphagov/pay-adminusers/pull/1229/commits/04175b226cc42a12fba8b21e7dbfd83d8b737f2c)

File: src/main/java/uk/gov/pay/adminusers/service/UserServices.java (hunk 1 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/adminusers/service/UserServices.java:282]: [Fixed](https://github.com/alphagov/pay-adminusers/pull/1229/commits/c071c9dbab19d7f7e0fab1c2c887efd023d5b874)

**Review by stephencdaly — APPROVED** [review]
Looks good

**Pull request merged**